### PR TITLE
Add colors to error message hints

### PR DESCRIPTION
**Pull request by ofonime41**
1. Created a ` hint` style tag and define the styles for the `Hint: `prefix.
2. Added the ` hint` tag to all the `Hint: `prefix.

### CODE REVIEW

File: utils/misc.ml [hunk 1]
```diff
@@ -623,6 +623,7 @@ module Color = struct
     | BG of color (* background *)
     | Bold
     | Reset
+    | Hint
```
Comment (Octachron) [inline utils/misc.ml]: This new constructor is unused: `Hint` is not a terminal style .

**Comment by Octachron** [comment]
There are many unrelated changes in the `Env` module that should be removed. You could also squash all your commits except the first one into one commit (after rebasing on trunk), it will help you to check that your commit changes are as minimal as possible.

**Comment by Octachron** [comment]
Was the PR closed by accident? If you have some git troubles, don't hesitate to ask for help.

**Comment by ofonime41** [comment]
I had to closed this PR because of the messed up commit. Currently working on another branched to have a cleaner commit history. I'll be opening another pull request before the end of the day on this. Thank you

**Comment by Octachron** [comment]
Rather than submitting a new PR, you can push a new `christy` branch with the simplified history to github and the PR should be updated automatically.

**Comment by ofonime41** [comment]
Alright, working on it.

**Comment by ofonime41** [comment]
PULL REQUEST REOPEND [HERE](https://github.com/ocaml/ocaml/pull/11685)

**Pull request closed without merging**